WordPress/WordPressApi/HTTPAuthenticationAlertController.swift

```diff
@@ -0,0 +1,106 @@
+import Foundation
+
+
+public class HTTPAuthenticationAlertController {
+
+    public typealias AuthenticationHandler = (NSURLSessionAuthChallengeDisposition, NSURLCredential?) -> Void
+
+    private static var onGoingChallenges = [NSURLProtectionSpace: [AuthenticationHandler]]()
+
+    static public func presentWithChallenge(challenge: NSURLAuthenticationChallenge, handler: AuthenticationHandler) {
+        if var handlers = onGoingChallenges[challenge.protectionSpace] {
+            handlers.append(handler)
+            onGoingChallenges[challenge.protectionSpace] = handlers
+            return
+        }
+        onGoingChallenges[challenge.protectionSpace] = [handler]
+
+        let  controller: UIAlertController
+        if challenge.protectionSpace.authenticationMethod == NSURLAuthenticationMethodServerTrust {
+            controller = controllerForServerTrustChallenge(challenge)
+        } else {
+            controller = controllerForUserAuthenticationChallenge(challenge)
+        }
+        guard var presentingController = UIApplication.sharedApplication().keyWindow?.rootViewController else {
+            return
+        }
+        if presentingController.presentedViewController != nil {
+            presentingController = presentingController.presentedViewController!
+        }
```

Comment:
I'm a bit concerned about this. I don't like traversing the VC hierarchy to present something, but I don't know what the right way is. What if the presented VC is already presenting another VC?